ext/gd/gd.c

```diff
@@ -3648,7 +3661,20 @@ static void php_image_filter_scatter(INTERNAL_FUNCTION_PARAMETERS)
 		colors = emalloc(num_colors * sizeof(int));
 
 		ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(hash_colors), color) {
-			*(colors + i++) = (int) zval_get_long(color);
+			bool failed = false;
+			ZVAL_DEREF(color);
+			zend_long tmp = zval_try_get_long(color, &failed);
+			if (failed) {
+				efree(colors);
+				zend_argument_value_error(5, "value must be of type int, %s given", zend_zval_type_name(color));
```

Comment:
Ditto type error and clarify value of the hash colors array

Reply:
Agreed